src/base/bittorrent/dbresumedatastorage.cpp

```diff
@@ -452,6 +452,16 @@ int BitTorrent::DBResumeDataStorage::currentDBVersion() const
 
 void BitTorrent::DBResumeDataStorage::createDB() const
 {
+    try
+    {
+        enableWALMode();
+    }
+    catch (const RuntimeError &err)
+    {
+        LogMsg(tr("Couldn't enable Write-Ahead Logging (WAL) journaling mode. Error: %1.")
+               .arg(err.message()), Log::WARNING);
+    }
+
     auto db = QSqlDatabase::database(DB_CONNECTION_NAME);
 
     const QWriteLocker locker {&m_dbLock};
```

Comment:
What does this locker protect? Should it protect the enableWALmode() too?

Reply:
> What does this locker protect?

There are two connections from separate threads are used so read write locking is for better concurrency purpose.
But it isn't really needed at this place since worker thread isn't initialized, so I removed it.